pkg/gatherers/conditional/validation.go

```diff
@@ -39,7 +39,7 @@ func validateGatheringRules(gatheringRules []GatheringRule) []error {
 	if !result.Valid() {
 		var errs []error
 		for _, err := range result.Errors() {
-			errs = append(errs, fmt.Errorf(err.String()))
+			errs = append(errs, fmt.Errorf("%s", err.String()))
```

Comment:
I presume this is done to copy the error because the `err` reference changes with each iteration. If that's the case, wouldn't it be better to use a similar approach to what we're currently doing with various K8s resource structs in the gathering code. Something like this:
```go
resultErrors := result.Errors()
for i := range resultErrors {
    errs = append(errs, resultErrors[i])
}
```
If the reason for this conversion from error to string and back to error is done for a different reason, then it's perfectly fine, but it just felt like unnecessary type gymnastics to me.

Reply:
Well... If the `resultErrors[i]` was actually an `error` that would be indeed possible, but it is more the case, the `resultErrors[i]` will return a `ResultError`.

Reply:
@rluders Oh, my bad, sorry about that. I didn't check the types, so I thought it was just regular go `error`s. In that case, we can't really do much about it. I believe there's also a way to directly convert a `string` into `error` somehow without having to do this formatting (should be slightly faster and a bit cleaner imo). Using `errors.New(string)` is one possibility that I have seen recommended, but if you don't feel like changing this, it's fine by me. I can approve the PR either way.

Reply:
I prefer to keep it as `fmt.Errorf` 'cause we are using it in several other places. I considered using the `errors.New()` but... TBH, didn't see any reasonable benefit. Lint is happy with both.

Reply:
Okay, sure, it's fine by me.